bitcoin/src/blockdata/transaction.rs

```diff
@@ -1084,23 +1092,27 @@ impl InputWeightPrediction {
     /// This is a `const` version of [`new`](Self::new) which only allows slices due to current Rust
     /// limitations around `const fn`. Because of these limitations it may be less efficient than
     /// `new` and thus is intended to be only used in `const` context.
-    pub const fn from_slice(input_script_len: usize, witness_element_lengths: &[usize]) -> Self {
+    pub const fn from_slice(input_script_len: u32, witness_element_lengths: &[u32]) -> Self {
         let mut i = 0;
-        let mut total_size = 0;
+        let mut total_size: u32 = 0;
         // for loops not supported in const fn
         while i < witness_element_lengths.len() {
             let elem_len = witness_element_lengths[i];
-            let elem_size = elem_len + compact_size::encoded_size_const(elem_len as u64);
-            total_size += elem_size;
+            let encoded_size = compact_size::encoded_size_const(elem_len as u64) as u32;
```

Comment:
Nope, you're just silently losing data here.

Reply:
How is this silently losing data?  `encoded_size_const` can be at most 9, so we could cast as u4 even if there is such a thing.

```rust
#[inline]
pub const fn encoded_size_const(value: u64) -> usize {
    match value {
        0..=0xFC => 1,
        0xFD..=0xFFFF => 3,
        0x10000..=0xFFFFFFFF => 5,
        _ => 9,
    }
}
```

Reply:
Oh, sorry I misread it as element length directly. `encoded_size_const` MUST take `u64` but should return `u8`. It'll not help on it's own but perhaps we can have `const fn u8_to_usize` & co in `internals` to help with these const casts.